src/V3Tristate.cpp

```diff
@@ -403,6 +403,20 @@ class TristateVisitor final : public TristateBaseVisitor {
         }
         return VN_AS(invarp->user1p(), Var);
     }
+    void replaceVarInExpressionWithEnVar(AstNode* nodep) {
+        // Replace variale references with master __en variable references
+        if (AstVarRef* varrefp = VN_CAST(nodep, VarRef)) {
+            AstVarRef* enVarrefp = new AstVarRef(varrefp->fileline(),
+                                                 getCreateEnVarp(varrefp->varp()), VAccess::READ);
+            varrefp->replaceWith(enVarrefp);
+        } else if (AstExtend* extendp = VN_CAST(nodep, Extend))
+            replaceVarInExpressionWithEnVar(extendp->lhsp());
+        else if (AstSel* selp = VN_CAST(nodep, Sel))
+            replaceVarInExpressionWithEnVar(selp->fromp());
+        else
+            nodep->v3warn(E_UNSUPPORTED,
```

Comment:
I think they'll want a test for this.

Reply:
There is test for Select. Or do you mean test for Unsupported?

Reply:
Unsupported

Reply:
Done